lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/LazyShapeDtype.cpp

```diff
@@ -0,0 +1,31 @@
+/**
+ * This is a handwritten file that accompanies codegenerated header
+ * LazyShapeDtype.h
+ *
+ * The purpose of these shape/dtype inference methods are to fill gaps
+ * where we do not yet have structured kernels in pytorch core.  Ops
+ * for which there _are_ structured kernels can use meta::op() to infer
+ * shape/dtype, and codegen makes use of this.  Ops for which there are not
+ * yet structured kernels can still be used with lazy_tensor codegen, but require
+ * manual intervention to implement compute_shape_{op} and compute_dtype_{op}.
+ *
+ */
+#include "lazy_tensor_core/csrc/ts_backend/LazyShapeDtype.h"
+
+namespace torch_lazy_tensors{
+namespace ir {
+namespace ops {
+
+std::vector<int64_t> compute_shape_mean(const at::Tensor& self, c10::optional<at::ScalarType> dtype) {
+    return std::vector<int64_t>({1});
```

Comment:
this will be updated in future to support cases where we don't reduce the whole tensor??

Reply:
i think that case is disambiguated by having a different set of function arguments (maybe even name?). -- so we'd generate another `compute_shape_mean` function which included the `dim` argument.

note: for some reason `mean` by itself has no meta kernel yet, but i think mean.dim does have a meta kernel.  so for mean.dim, we'd never have to provide this function, just rely on meta.